### PR TITLE
Remove callStructDtorsDuringGC rt flag

### PR DESCRIPTION
**Pull request by edi33416**
Struct dtors should always be called; we shouldn't allow configuration files to change the meaning of the language.

Edit: Anyone knows when and why was this introduced? Any projects using it?

### CODE REVIEW

**Comment by thewilsonator** [comment]
This i separate from the not calling the GC at all on shutdown, right?

**Comment by edi33416** [comment]
> This i separate from the not calling the GC at all on shutdown, right?

As far as I can tell, this is only used by `structTypeInfoSize` to determine if `_d_newitem` needs to store a ptr to the `dtor`. I don't think this is related with the GC shutdown.

**Comment by andralex** [comment]
I wonder how to deprecate this - issue a message at runtime to stderr if the flag is false?

**Comment by Geod24** [comment]
> Edit: Anyone knows when and why was this introduced? Any projects using it?

A long time ago, the GC would not call destructors on heap allocated struct. When it was fixed, I think this was put as a transitional step. https://github.com/dlang/druntime/pull/864

File: src/rt/lifetime.d [hunk 1]
```diff
@@ -2590,7 +2578,6 @@ unittest
     delete arr1;
     assert(dtorCount == 7);
 
-    if (callStructDtorsDuringGC)
```
Comment (rainers) [inline src/rt/lifetime.d]: Please remove scoping/indentation, too. Later it will just confuse people why it is there sometimes, and sometimes not.

**Comment by jacob-carlborg** [comment]
> I wonder how to deprecate this - issue a message at runtime to stderr if the flag is false?

I would print the message as long as the flag has been passed to the application, regardless of the value of the flag.

**Comment by jacob-carlborg** [comment]
> AFAICT it has never been documented so we might consider it internal and skip a deprecation, or make that a short one.

Are any of the druntime flags documented?

**Comment by wilzbach** [comment]
> > AFAICT it has never been documented so we might consider it internal and skip a deprecation, or make that a short one.
> 
> Are any of the druntime flags documented?

No. "Luckily", druntime is chronically under-documented.
The only documentation of `--DRT` is here: https://dlang.org/spec/garbage.html#gc_config (for the GC bits).

**Comment by jacob-carlborg** [comment]
I think it should be deprecated anyway. It's not difficult to do. For me it's enough for it to be deprecated for one release.

**Comment by JohanEngelen** [comment]
'// this is run before static ctors, so it is safe to modify immutables'

Wow, incredible. Very nice to get rid of this kind of UB in druntime code.

**Comment by dnadlinger** [comment]
> Wow, incredible. Very nice to get rid of this kind of UB in druntime code.

Immutable globals can be initialised in static constructors, so whether this is UB or not is questionable.

**Comment by andralex** [comment]
@edi33416 ?

**Comment by edi33416** [comment]
@andralex @jacob-carlborg I opened #2513 for the deprecation of the feature

**Pull request closed without merging**